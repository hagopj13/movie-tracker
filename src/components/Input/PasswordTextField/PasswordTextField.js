// @flow
import React, { useState } from 'react';
import TextField from '@material-ui/core/TextField';
import InputAdornment from '@material-ui/core/InputAdornment';
import IconButton from '@material-ui/core/IconButton';
import VisibilityIcon from '@material-ui/icons/Visibility';
import VisibilityOffIcon from '@material-ui/icons/VisibilityOff';

const PasswordTextField = (props: any) => {
  const [showPassword, setShowPassword] = useState(false);

  const handleShowPasswordClick = () => setShowPassword(!showPassword);

  const handleShowPasswordMouseDown = (event) => event.preventDefault();

  return (
    <TextField
      {...props}
      type={showPassword ? 'text' : 'password'}
      InputProps={{
        endAdornment: (
          <InputAdornment position="end">
            <IconButton
              onClick={handleShowPasswordClick}
              onMouseDown={handleShowPasswordMouseDown}
              edge="end"
            >
              {showPassword ? <VisibilityIcon /> : <VisibilityOffIcon />}
            </IconButton>
          </InputAdornment>
        ),
      }}
    />
  );
};

export default PasswordTextField;
