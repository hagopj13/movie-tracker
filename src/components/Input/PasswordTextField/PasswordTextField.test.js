import React from 'react';
import { createShallow, createMount } from '@material-ui/core/test-utils';
import TextField from '@material-ui/core/TextField';
import IconButton from '@material-ui/core/IconButton';
import VisibilityIcon from '@material-ui/icons/Visibility';
import VisibilityOffIcon from '@material-ui/icons/VisibilityOff';

import PasswordTextField from './PasswordTextField';

describe('PasswordTextField component', () => {
  let shallow;
  let mount;

  beforeAll(() => {
    shallow = createShallow();
    mount = createMount();
  });

  it('should render the PasswordTextField component correctly', () => {
    const wrapper = shallow(<PasswordTextField />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should initially set the textfield type to password', () => {
    const wrapper = shallow(<PasswordTextField />);
    expect(wrapper.find(TextField).prop('type')).toBe('password');
  });

  it('should initially show the visibility off icon', () => {
    // use mount here to make sure endAdornment renders
    const wrapper = mount(<PasswordTextField />);
    expect(wrapper.exists(VisibilityOffIcon)).toBe(true);
  });

  it('should set the textfield type to text after show password icon is clicked', () => {
    // use mount here to make sure endAdornment renders
    const wrapper = mount(<PasswordTextField />);
    wrapper.find(IconButton).simulate('click');
    expect(wrapper.find(TextField).prop('type')).toBe('text');
  });

  it('should initially show the visibility off icon', () => {
    // use mount here to make sure endAdornment renders
    const wrapper = mount(<PasswordTextField />);
    wrapper.find(IconButton).simulate('click');
    expect(wrapper.exists(VisibilityIcon)).toBe(true);
  });
});
