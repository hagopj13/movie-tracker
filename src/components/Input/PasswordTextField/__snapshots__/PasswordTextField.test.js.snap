// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordTextField component should render the PasswordTextField component correctly 1`] = `
<WithStyles(ForwardRef(TextField))
  InputProps={
    Object {
      "endAdornment": <ForwardRef(WithStyles)
        position="end"
      >
        <ForwardRef(WithStyles)
          edge="end"
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <UNDEFINED />
        </ForwardRef(WithStyles)>
      </ForwardRef(WithStyles)>,
    }
  }
  type="password"
/>
`;
